File: Coding Challenges/DailyChallenge/JavaScript/20260120_ConsonantCase.js
function toConsonantCase(str) {
  const vowels = /[aeiou]/gi;
  const consonants = /[bcdfghjkklmnopqrstvwxyz]/gi
  
  let newString = '';

  for(let i = 0; i < str.length; i++) {
    let char = str[i];
    
    if(char.search(vowels) != -1) {
      newString += char.toLowerCase();
    }
    else if(char.search(consonants) != -1) {
      newString += char.toUpperCase();
    }
    else if(char == '-') {
      newString += '_';
    }
    else {
      newString += char;
    }
  }
  
  return newString;
}

File: Coding Challenges/DailyChallenge/Python/20260120_ConsonantCase.py
import re

def to_consonant_case(s):
    vowels = '[aeiou]'
    consonants = '[bcdfghjkklmnopqrstvwxyz]'

    new_string = ''

    for char in s:
        if re.search(vowels, char, re.IGNORECASE):
            new_string += char.lower()
        elif re.search(consonants, char, re.IGNORECASE):
            new_string += char.upper()
        elif char == '-':
            new_string += '_'
        else:
            new_string += char

    return new_string